runbotics/runbotics-scheduler/src/mail/mail.service.ts

```diff
@@ -0,0 +1,203 @@
+                }
+            }
+
+            if (processedMails.length) {
+                this.logger.log('Mails processsed: ', processedMails);
+            } else {
+                this.logger.log('No mails processed ');
+            }
+
+            await this.client.messageFlagsAdd(processedMails, ['\\Seen']);
+        } finally {
+            await this.client.mailboxClose();
+            lock.release();
+            await this.client.logout();
+            this.logger.log('<< Mailbox disconnected');
+        }
+    }
+
+    private async validateTitle(envelope: MessageEnvelopeObject) {
+        const process = await this.processService.findByInfo(envelope.subject);
+
+        if (!process) {
+            this.logger.error(`Process "${envelope.subject}" does not exist`);
+            throw new NotFoundException(`Process "${envelope.subject}" does not exist`);
+        }
+
+        await this.queueService.validateProcessAccess({ process, triggered: true });
+
+        return process;
+    }
+
+    private sendReplyErrorMail(envelope: MessageEnvelopeObject, text: string) {
+        return this.server.sendMail({
+            from: envelope.to[0].address,
+            to: envelope.from[0].address,
+            replyTo: envelope.replyTo[0].address,
+            inReplyTo: envelope.messageId,
+            subject: envelope.subject,
+            text: `Error running process ${envelope.subject}\n\n${text}`,
+        });
+    }
+
+    private async extractMailBody(body: Buffer) {
+        try {
+            const parsed = await simpleParser(body);
+            const input = this.parseMailBody(parsed.text);
+            return {
+                ...input,
+                ...this.extractAttachments(parsed.attachments)
+            };
+        } catch (error) {
+            this.logger.error(`Wrong body format: ${error.message}`);
+            throw new Error(`Wrong body format: ${error.message}`);
+        }
+    }
+
+    private extractAttachments(attachments: Attachment[]) {
+        return attachments.reduce((acc, att) => {
+            const filename = att.filename.split('.').slice(0, -1).join('');
+            acc[filename] = `data:${att.contentType};base64,'${att.content.toString('base64')}`;
+            return acc;
+        }, {});
+    }
+
+    private parseMailBody(text: string) {
+        return text
+            .split(/\r?\n/)
+            .reduce((input, line, index) => {
+                if (!line) return input;
+                // eslint-disable-next-line @typescript-eslint/no-unused-vars
+                const [key, value, _] = line.split('=');
+                if (!key || !value)
+                    throw new Error(`Invalid variable format at line ${index + 1}`);
+                input[key.trim()] = value.trim().replace(/['"]/g, '');
+                return input;
+            }, {});
+    }
+
+    private hasTriggerAccess(emailAdress: string) {
+        return emailAdress.includes('@all-for-one.com');
```

Comment:
hm, maybe we can create something more reusable, like array with allowed domains stored in env?